Mage.Sets/src/mage/cards/g/Glamdring.java

```diff
@@ -0,0 +1,76 @@
+package mage.cards.g;
+
+import mage.abilities.Ability;
+import mage.abilities.common.DealsDamageToAPlayerAttachedTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.dynamicvalue.common.CardsInControllerGraveyardCount;
+import mage.abilities.dynamicvalue.common.StaticValue;
+import mage.abilities.dynamicvalue.common.TriggerDamageDone;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.common.continuous.BoostEquippedEffect;
+import mage.abilities.effects.common.continuous.GainAbilityAttachedEffect;
+import mage.abilities.effects.common.cost.CastFromHandForFreeEffect;
+import mage.abilities.keyword.EquipAbility;
+import mage.abilities.keyword.FirstStrikeAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.FilterCard;
+import mage.filter.StaticFilters;
+import mage.filter.common.FilterCompareManaValue;
+import mage.filter.predicate.Predicates;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author bobby-mccann
+ */
+public final class Glamdring extends CardImpl {
+    private static final FilterCard filter = new FilterCompareManaValue(ComparisonType.FEWER_THAN_OR_EQUAL_TO, new TriggerDamageDone());
+    static {
+        filter.add(Predicates.or(
+                CardType.INSTANT.getPredicate(),
+                CardType.SORCERY.getPredicate()
+        ));
+        filter.setMessage("an instant or sorcery spell from your hand with mana value less than or equal to that damage");
```

Comment:
This needs reworked overall as per my comments on your other new classes, and should be able to set filter message directly in the constructor.